assemble/src/main/scripts/generate-download-script.sh

```diff
@@ -0,0 +1,56 @@
+#! /usr/bin/env bash
+
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script will generate a DEPENDENCIES listing of packaged dependencies
+
+in=target/dependencies.raw.txt
+out=target/download-dependencies
+
+cat >"$out" <<'EOF'
+#! /usr/bin/env bash
+# This script downloads the following jars, identified by their maven
+# coordinates, using the maven-dependency-plugin.
+#
+# DISCLAIMER: This is only one possible way to download a set of dependencies
```

Comment:
> A user might look at versions, see that they want 0.9.4 Thrift for some security patch or for other things in their application.

That poor user.  I have seen users struggle with Thrift.  One situation is the user wants to use Accumulo client (where Accumulo client depends on thrift Z) and library X (where X depends on thrift ver Y) in the same JVM.  With libthrifts complete lack of comparability guarantees this is not a good situation to be in. I think I have built a shaded jar w/ libthrift relocted to get around this type of thing.